src/modules/dashboard/components/Members/filterMembers.ts

```diff
@@ -1,21 +1,64 @@
+import { MEMEBERS_FILTERS } from '~dashboard/ColonyMembers/MembersFilter';
+
 import { ColonyContributor, ColonyWatcher } from '~data/index';
 
 export const filterMembers = <M extends ColonyContributor | ColonyWatcher>(
   data: M[],
-  filterValue: string,
+  filterValue?: string,
+  filtersList?: MEMEBERS_FILTERS[],
 ): M[] => {
-  if (!filterValue) {
+  const excludeBanned = !filtersList?.includes(MEMEBERS_FILTERS.BANNED);
+  const excludeVerified = !filtersList?.includes(MEMEBERS_FILTERS.VERIFIED);
+
+  /* No filters */
+  if (!filterValue && !excludeBanned && !excludeVerified) {
     return data;
   }
 
-  return data.filter(
-    (member) =>
-      member?.profile?.username
-        ?.toLowerCase()
-        .includes(filterValue.toLowerCase()) ||
-      member?.profile?.walletAddress
+  /* Only text filter */
+  if (filterValue && !excludeBanned && !excludeVerified) {
+    return data.filter(
+      ({ profile, id }) =>
+        profile?.username?.toLowerCase().includes(filterValue.toLowerCase()) ||
+        profile?.walletAddress
+          ?.toLowerCase()
+          .includes(filterValue.toLowerCase()) ||
+        id?.toLowerCase().includes(filterValue.toLowerCase()),
+    );
+  }
+
+  /* Only checkbox filters */
+  if (!filterValue) {
+    return data.filter(({ banned, isWhitelisted }) => {
+      if (excludeBanned && excludeVerified) {
+        return !banned && !isWhitelisted;
+      }
+
+      if (excludeBanned) {
+        return !banned;
+      }
+
+      return !isWhitelisted;
+    });
+  }
+
+  /* All the filters together */
+  return data.filter(({ banned, isWhitelisted, profile, id }) => {
+    const textFilter =
+      profile?.username?.toLowerCase().includes(filterValue?.toLowerCase()) ||
```

Comment:
This block is almost a duplicate of the above, just without `textFilter`. Could this be refactored?